src/components/pages/home/Home.js

```diff
@@ -3,24 +3,88 @@ import React, { useState, useEffect } from 'react'
 import Products from '../../products/Products'
 import SearchBar from '../../searchbar/Searchbar'
 import itemsServices from '../../../services/items'
+import PaginationButton from '../../PaginationButtons/PaginationButton'
 
 import './home.css'
 
+const PAGE_SIZE = 6;
+
 const Home = () => {
     const [items, setItems] = useState([])
+    const [noItems, setNoItems] = useState(false)
+    const [number_of_pages, setNumberOfPages] = useState(null)
+    const [page, setPage] = useState(1)
+    const [search, setSearch] = useState(false)
+
+    // fetches for all the data in the backend
+    const fetch_data = async () => {
+        // Calculate the "from" of the query params 
+        const request_from = (page - 1) * PAGE_SIZE
+        // Get all the ietems of the backend to calculate number of pages
+        const response_data = await itemsServices.getAllItems()
+        // Set number of pages in pagination
+        setNumberOfPages(Math.ceil(response_data.items.length/PAGE_SIZE))
+
+        const data_items = await itemsServices.getPageItems(request_from)
+        setItems(data_items.items)
+    }
 
+    // eslint-disable-next-line
     useEffect(() => {
-        const fetch_data = async () => {
-            const response_data = await itemsServices.getAllItems()
-            setItems(response_data.items)
-        }
         fetch_data()        
-    }, [])
+    }, [page])
+
+    // called when a user does a search with the searchBar component
+    const handleSearch = async (searchItem) => {
+        if (searchItem === '') {
+            setPage(1)
+            setNoItems(false)
+            setSearch(false)
+            fetch_data()
```

Comment:
This is fine, but it is better practice to run this in a useEffect when either `page`, or `searchText` state value changes